themes/default/content/docs/using-pulumi/adopting-pulumi/import/_index.md

```diff
@@ -12,161 +12,184 @@ aliases:
 
-When importing resources via the CLI, there is a specific import syntax definition required for each resource. To see this definition, navigate to the corresponding resource page in the Pulumi registry and take a look at the **Import** section. Using the [AWS S3 Bucket resource](https://www.pulumi.com/registry/packages/aws/api-docs/s3/bucket/#import) as an example, the following command imports an existing AWS S3 bucket named `company-infra-logs` and defines the resource name for the Pulumi program as `infra-logs`:
+### Example
+
+In this example, a previously provisioned Amazon S3 bucket named `company-infra-logs` is imported into a Pulumi stack named `dev` (the currently [selected](/docs/cli/commands/pulumi_stack_select/) stack) and given a resource name of `infra-logs`:
 
 ```bash
 $ pulumi import aws:s3/bucket:Bucket infra-logs company-infra-logs
 
-     Type                 Name             Plan
- +   pulumi:pulumi:Stack  import-post-dev  create
- =   └─ aws:s3:Bucket     infra-logs       import
+Previewing import (dev)
+
+     Type                 Name        Plan
+ +   pulumi:pulumi:Stack  dev         create
+ =   └─ aws:s3:Bucket     my-bucket   import
 
 Resources:
     + 1 to create
     = 1 to import
     2 changes
+
+Do you want to perform this import?
+> yes
+  no
+  details
 ```
 
-Pulumi will perform the import of the S3 bucket and generate the code required for you to add it to your
-application.
+Choosing to proceed immediately adds the resource to the current stack's state and emits a block of code to `STDOUT` to be added to the Pulumi program. If the current program were written in TypeScript, for example, the resulting CLI output would resemble the following:
 
-{{< chooser language "typescript,python,csharp,go" >}}
+```plain
+...
 
-{{% choosable language typescript %}}
+Importing (dev)
+
+     Type                 Name        Status
+ +   pulumi:pulumi:Stack  dev         created (3s)
+ =   └─ aws:s3:Bucket     my-bucket   imported (1s)
+
+Resources:
+    + 1 created
+    = 1 imported
+    2 changes
+
+Duration: 4s
+
+Please copy the following code into your Pulumi application. Not doing so
+will cause Pulumi to report that an update will happen on the next update command.
+
+Please note that the imported resources are marked as protected. To destroy them
+you will need to remove the `protect` option and run `pulumi update` *before*
+the destroy will take effect.
 
-```typescript
 import * as pulumi from "@pulumi/pulumi";
 import * as aws from "@pulumi/aws";
 
-const demo = new aws.s3.Bucket("infra-logs", {
-    acl: "private",
+const my_bucket = new aws.s3.Bucket("infra-logs", {
+    arn: "arn:aws:s3:::company-infra-logs",
```

Comment:
It really returns the ARN? I thought that was strictly an output.

Reply:
It does! This is the actual output, even on a brand-new empty bucket. I actually wonder whether it's worth filing an issue on this, as you can remove everything except `bucket`, run `up`, and get no changes. What do you think?